### PR TITLE
Add start/stop support for flinkclusters

### PR DESCRIPTION
**Pull request by keymone**

### CODE REVIEW

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 1]
```diff
@@ -107,12 +110,20 @@ def log_event(service_config, desired_state):
...
-        force_bounce=force_bounce,
-        desired_state=desired_state,
-    )
-    remote_git.create_remote_refs(utils.get_git_url(service_config.get_service()), ref_mutator)
+    if isinstance(service_config, FlinkClusterConfig):
+        set_flinkcluster_desired_state(
+            kube_client=KubeClient(),
+            service=service_config.service,
+            instance=service_config.instance,
+            desired_state=dict(start='running', stop='stopped')[desired_state],
```
Comment (vkhromov) [inline paasta_tools/cli/cmds/start_stop_restart.py]: lol

File: paasta_tools/cli/cmds/start_stop_restart.py [hunk 2]
```diff
@@ -147,6 +158,15 @@ def print_chronos_message(desired_state):
         )
 
 
+def print_flinkcluster_message(desired_state):
+    if desired_state == "start":
+        # paasta_print()
```
Comment (vkhromov) [inline paasta_tools/cli/cmds/start_stop_restart.py]: please actually print something here and below

File: paasta_tools/flinkcluster_tools.py [hunk 3]
```diff
@@ -79,3 +79,42 @@ def get_flinkcluster_config(
...
     return status
+
+
+def set_flinkcluster_desired_state(
+    kube_client: KubeClient,
+    service: str,
+    instance: str,
+    desired_state: str,
+):
+    sanitised_service = service.replace('_', '--')
```
Comment (vkhromov) [inline paasta_tools/flinkcluster_tools.py]: use `sanitise_name()`

File: paasta_tools/flinkcluster_tools.py [hunk 4]
```diff
@@ -79,3 +79,42 @@ def get_flinkcluster_config(
...
+        version=version,
+        namespace=namespace,
+        plural=plural,
+    )
+    if 'metadata' not in co:
+        co['metadata'] = {}
+    if 'annotations' not in co['metadata']:
+        co['metadata']['annotations'] = {}
+    co['metadata']['annotations']['yelp.com/desired_state'] = desired_state
+    kube_client.custom.replace_namespaced_custom_object(
```
Comment (vkhromov) [inline paasta_tools/flinkcluster_tools.py]: can you please double check that it updates the object and not deletes/creates it?

**Comment by Rob-Johnson** [comment]
ship once you've tested on marathon/chronos

**Pull request merged**